src/metatrain/utils/evaluate_model.py

```diff
@@ -107,41 +107,28 @@ def evaluate_model(
         systems = new_systems
     else:
         if len(energy_targets_that_require_position_gradients) > 0:
-            if not is_exported(model):
-                new_systems = []
-                for system in systems:
-                    new_system = System(
-                        positions=system.positions.detach()
-                        .clone()
-                        .requires_grad_(True),
-                        cell=system.cell,
-                        types=system.types,
+            new_systems = []
```

Comment:
I think this part of code would be cleaner if split differently. Maybe something closer to:

```
def prepare_system(system, positions_grad=False, strain_grad=False):
    if strain_grad:
        strain = torch.eye(3, dtype=..., device=...)

        positions = system.positions @ strain
        if positions_grad:
            positions.retain_grad()

        cell = system.cell @ strain
    else:
        strain = None

    system = System(
        types=system.types,
        positions=positions,
        cell=cell,
    )

    # compute all NL here

    return system, strain
```

This way we only need to compute NL once, regardless of the different gradient options